### DISCUSSION TITLE
Can't set model as property if it hasn't been persisted yet

### DISCUSSION THREAD

**Discussion by RebelSoftware** [livewire/livewire#6511] [7. Bugs (archived)]
### Livewire version

3.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Create a component

create a public property based on a Model

render the component with an new model (not persisted)

For some reason this is not allowed?

I have a model called Flight,

If I do this it fails

```
class TestModel extends Component
{
    public Flight $flight;
    public function mount() {
        $this->flight = new Flight();
    }
...
}

```
But this works fine

```
class TestModel extends Component
{
    public $flight;
    public function mount() {
        $this->flight = Flight::find(1);
    }
...
}

```

While I am aware of the legacy_model_binding config setting I cannot understand why a model has to be persisted before it can be used.

My intention is to create a new Flight model and create it or update it in the Save() function.

However I need the model to access attributes of the model while rendering the livewire component so I need to instantiate an empty Model to access those attributes.

If this is not a bug per se can you please explain the thought process behind disallowing models that do not exist in the DB (or wherever)?

**EDIT:** I forgot to mention this is an upgraded install from V2+

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by N1ebieski** [comment]
[Try this](https://github.com/livewire/livewire/blob/main/config/livewire.php#L92-L103)

**Reply by RebelSoftware** [reply]
As stated I am aware of that setting. 

The description from that page says that the binding is too "magical". yet it is both default Laravel behaviour to instantiate a model if used as a parameter and not relevant in this situation, as I am not using it as a parameter. it is a predefined property.

**Reply by RebelSoftware** [reply]
Also It feels a bit hacky to me to use legacy settings to make a new version work. Which is why I'd prefer to know why it considered necessary. 

The description that comes with legacy_model_binding makes it appear as if its more of an oversight (nee bug) than a deliberate attempt to prevent a Model from being instantiated deliberately.
